### ISSUE TITLE
Upgrade to docker-compose.yml version 2

### ISSUE THREAD

**Issue by justinlittman** [gwu-libraries/sfm-ui#382]
Version 1 is increasingly deprecated.

**Comment by justinlittman** [comment]
Here's what I've learned that must be changed:
* Add version 2 declaration and add `services`:
```
version: "2"
services:
    sfmdb:
```
* Builds must be changed:
```
        build:
            context: ../../sfm-wayback/docker/wayback
```
* Linked environment variables are deprecated.  Environment variables must be explicitly set.  This can be done in combination with a `.env` file to avoid duplication.
```
        environment:
            - TZ=${TZ}
            - DEBUG=True
            - MQ_ENV_RABBITMQ_DEFAULT_USER=${RABBITMQ_DEFAULT_USER}
            - MQ_ENV_RABBITMQ_DEFAULT_PASS=${RABBITMQ_DEFAULT_PASS}
```

**Comment by justinlittman** [comment]
Make sure to test processing containers, as may need to add `--net sfmtwitterharvester_default` or similar to command.

**Comment by justinlittman** [comment]
Other docker options to consider as part of work on this ticket:
* Get rid of example docker-compose files in every project except sfm-docker.
* Add an example dev docker-compose.yml file to sfm-docker.
* Address #360 (naming docker containers).
* Move configuration into .env or secrets.env as suggested by #411.

**Comment by justinlittman** [comment]
To consider:
1.  Creating a base image for harvesters, exporters, etc.  (Look at ONBUILD command, ENV substitution).
2.  Creating a single master docker-compose.yml file.  Replace image versions, etc. with variables.  Also, use a docker-compose.override.yml file.  Or use extends command.
3.  Using multiple docker-compose.yml files to handle extension containers like ELK or processing containers.  (-f can be used to provide a list of docker-compose files.)

**Comment by lwrubel** [comment]
Closed with #418
